crategen/converters/tes_converter.py

```diff
@@ -1,52 +1,66 @@
 from .abstract_converter import AbstractConverter
 from .utils import convert_to_iso8601
+from ..models import TESData, WRROCDataTES
+from pydantic import ValidationError
 
 class TESConverter(AbstractConverter):
 
     def convert_to_wrroc(self, tes_data):
-        # Validate and extract data with defaults
-        id = tes_data.get("id", "")
-        name = tes_data.get("name", "")
-        description = tes_data.get("description", "")
-        executors = tes_data.get("executors", [{}])
-        inputs = tes_data.get("inputs", [])
-        outputs = tes_data.get("outputs", [])
-        creation_time = tes_data.get("creation_time", "")
-        end_time = tes_data.get("logs", [{}])[0].get("end_time", "")  # Corrected to fetch from logs
+        # Validate TES data
+        try:
+            validated_tes_data = TESData(**tes_data)
+        except ValidationError as e:
+            raise ValueError(f"Invalid TES data: {e}")
+
+        # Extract validated data
+        id = validated_tes_data.id
+        name = validated_tes_data.name
+        description = validated_tes_data.description
+        executors = validated_tes_data.executors
+        inputs = validated_tes_data.inputs
+        outputs = validated_tes_data.outputs
+        creation_time = validated_tes_data.creation_time
+        end_time = validated_tes_data.logs[0].end_time if validated_tes_data.logs else ""
 
         # Convert to WRROC
         wrroc_data = {
             "@id": id,
             "name": name,
             "description": description,
-            "instrument": executors[0].get("image", None) if executors else None,
-            "object": [{"@id": input.get("url", ""), "name": input.get("path", "")} for input in inputs],
-            "result": [{"@id": output.get("url", ""), "name": output.get("path", "")} for output in outputs],
+            "instrument": executors[0].image if executors else None,
```

Comment:
I've noted the concern regarding multiple executors. I've opened an issue to address how WRROC entities should handle multiple executors. This will be addressed in a future PR.